### DISCUSSION TITLE
Generic OIDC flow with fief python client

### DISCUSSION THREAD

**Discussion by francbartoli** [fief-dev/fief#69] [Q&A]
Are generic OpenID Connect compliant identity providers like OpenAM supported by the fief client package? I have configured the `FiefAsync` class and followed the tutorial for FastAPI integration but I'm getting a NetworkError like in the picture when I land back
<img width="1674" alt="Senza nome 11" src="https://user-images.githubusercontent.com/560676/188455966-3a5e7a57-43aa-4285-b5d6-29afaad3130d.png">

**Comment by frankie567** [comment]
Hello @francbartoli 👋 

The Fief client is designed to work with Fief 🙃

I'm not sure to understand what you're trying to do here. I don't know OpenAM but it seems to be similar to Fief. Do you want to use Fief in conjunction with OpenAM?

**Reply by francbartoli** [reply]
Hi @frankie567, thanks for your response. I like fief with the decoupling of server/client and since OpenID Connect is a standard flow I expected to be working with different server implementations. I tried with OpenAM but Keycloak is another example.
Also, as far as I can see fief-client is likely the only one to support the PKCE mode within a FastAPI application and I need it for a project. I can work to see if this support might be widely available, is it of interest to you?

**Reply by frankie567** [reply]
Hmm, I see! Well there are two things here:

1. Regarding the OpenID/OAuth2 protocol, I think the client will work quite well for any compliant provider. If you give the right base URL, I think it'll be able to redirect to the authentication page and get a fresh token on callback without any issue.
2. However, concerning the `authenticated` dependency, I'm really not sure it'll work with other systems: it greatly depends how they implemented their access token.
    * In Fief, we just a [JWT](https://docs.fief.dev/getting-started/oauth2/#access-token-and-id-token), meaning we can check its validity locally, without a roundtrip to the authentication server.
    * Besides , permissions are directly included in the data of the JWT, that's how we can check them directly.
    * If OpenAM, Keycloak or others don't do that, the `authenticated` dependency will reject the token.

For the basic OpenID protocol, I suggest you look at [httpx-oauth](https://frankie567.github.io/httpx-oauth/oauth2/#openid), which is another of my project implementing the OAuth2 protocol for getting access token.
